### PR TITLE
Escape behavior updated and fixed 

### LINKED ISSUE #1116: Esc key should cancel edits instead of saving
## Steps to reproduce

1. Begin editing a text cell.
1. Modify the contents.
1. Press the Esc key.
1. Expect the cell to transition to select mode, restoring the value before you began making your edits.
1. Observe the cell to transition to select mode, saving the value you entered.

## Expected behavior
1. For stepped input columns (all string based columns, number etc.,) which contain an edit mode, "Esc" key in edit mode should revert the value to what is present in the DB and cancel unsaved edits.
1. For all other columns, "Esc" key would essentially do nothing since value change is based on different interactions.

## Additional context

- The current behavior was established in #1080.
- "Save of Esc" is consistent with AirTable.
- "Cancel on Esc" is consistent with LibreOffice and Google Sheets

### PR DESCRIPTION
**Pull request by chidam333**
Fixes #1116

Escape key cancels edits and transitions to select mode
**Technical details**
I have made modifications to

- Number cell
- Textareacell
- Textcell
- Uricell

I had prblms in git so i went with manually uploading files ,but all linting and tests where passed .

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `master` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar_ui/src/components/cell-fabric/data-types/components/number/NumberCell.svelte [hunk 1]
```diff
@@ -38,6 +38,25 @@
     maximumFractionDigits,
   });
 
+<<<<<<< HEAD
```
Comment (seancolsen) [inline mathesar_ui/src/components/cell-fabric/data-types/components/number/NumberCell.svelte:41]: Lines that begin with `<<<<<<<` or `=======` or `>>>>>>>` should not be in the diff. Those are from git merge conflicts.

**Review by seancolsen — CHANGES_REQUESTED** [review]
I've not given this a thorough review because I can tell immediately that it needs changes due to a botched git merge.

@chidam333 From your PRs and comments in Matrix, it seems that you're new to git. I know it can be a lot to learn at first, so I apologize that our core team doesn't have more bandwidth to help you out. We are pushing very hard right now towards our first release. While we're able to review PRs, we likely won't have much time for hand-holding.

I hope you're able to figure it out! Make sure to test your work locally and when you ask for help in Matrix keep your questions very focused by explaining: (1) exactly what you did, (2) what you expected, and (3) what you observed instead.

**Comment by chidam333** [comment]
Hello @seancolsen and community thank you all for the help , i find myself guilty not reading docs properly for issues like linting , i will surely try to optimise the resources i refer from  and ask better questions

**Comment by seancolsen** [comment]
I'm closing this PR because it seems to be superseded by #1827

**Pull request closed without merging**